python/dgl/data/gnn_benckmark.py

```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
+
+    @property
+    def num_classes(self):
+        """Number of classes."""
+        return 5
+
+
+class AmazonCoBuyPhotoDataset(GNNBenchmarkDataset):
+    r"""AmazonCoBuy dataset for node classification task.
+
+    .. deprecated:: 0.5.0
+        `data` is deprecated, it is repalced by:
+        >>> dataset = AmazonCoBuyPhotoDataset()
+        >>> graph = dataset[0]
+
+    Amazon Computers and Amazon Photo are segments of the Amazon co-purchase graph [McAuley et al., 2015],
+    where nodes represent goods, edges indicate that two goods are frequently bought together, node
+    features are bag-of-words encoded product reviews, and class labels are given by the product category.
+
+    Reference: https://github.com/shchur/gnn-benchmark#datasets
+
+    Statistics
+    ----------
+    Nodes: 7,650
+    Edges: 287,326
+    Number of classes: 5
+    Node feature size: 745
+
+    Parameters
+    ----------
+    raw_dir : str
+        Raw file directory to download/contains the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+        Whether to print out progress information. Default: True.
+
+    Attributes
+    ----------
+    num_classes : int
+        Number of classes for each node.
+
+    Examples
+    --------
+    >>> data = AmazonCoBuyPhotoDataset()
+    >>> g = data[0]
+    >>> num_class = data.num_classes
+    >>> feat = g.ndata['feat']  # get node feature
+    >>> label = g.ndata['label']  # get node labels
+    """
+    def __init__(self, raw_dir=None, force_reload=False, verbose=False):
+        super(AmazonCoBuyPhotoDataset, self).__init__(name='amazon_co_buy_photo',
+                                                      raw_dir=raw_dir,
+                                                      force_reload=force_reload,
+                                                      verbose=verbose)
+
+    @property
+    def num_classes(self):
+        """Number of classes."""
+        return 5
+
+
+class CoraFull(CoraFullDataset):
+    def __init__(self, **kwargs):
+        deprecate_class('CoraFull', 'CoraFullDataset')
+        super(CoraFull, self).__init__(**kwargs)
+
+
+def AmazonCoBuy(name):
+    if name == 'computers':
+        deprecate_class('AmazonCoBuy', 'AmazonCoBuyComputerDataset')
+        return AmazonCoBuyComputerDataset()
+    elif name == 'photo':
+        deprecate_class('AmazonCoBuy', 'AmazonCoBuyPhotoDataset')
+        return AmazonCoBuyPhotoDataset()
+    else:
+        raise ValueError('Dataset name should be "computers" or "photo".')
+
+
+def Coauthor(name):
+    if name == 'cs':
+        deprecate_class('Coauthor', 'CoauthorCSDataset')
+        return CoauthorCSDataset()
+    elif name == 'physics':
+        deprecate_class('Coauthor', 'CoauthorPhysicsDataset')
+        return CoauthorPhysicsDataset()
+    else:
+        raise ValueError('Dataset name should be "cs" or "physics".')
```

Comment:
Do we need to update some examples ?

Reply:
There are no examples or tutorials using these datasets.